google_guest_agent/main.go

```diff
@@ -225,7 +225,12 @@ func run(ctx context.Context) {
 
 		// Make sure we close the pipe after we've done writing to it.
 		pipeData := evData.Data.(*sshtrustedca.PipeData)
-		defer pipeData.File.Close()
+		defer func() {
+			if err := pipeData.File.Close(); err != nil {
+				logger.Errorf("Failed to close pipe: %+v", err)
+			}
+			pipeData.Finished()
```

Comment:
I think we should try to keep this call in sshtreustedca's Run function, otherwise we're relying on a call to `Finished` that could never come leading to an infinite loop.

Reply:
This is an "async call", we can't exactly call it from `Run()`, think of it as a bottom half strategy but with a sync up requirement.